### PR TITLE
[IOTDB-3648] Reinforce ConfigNode startup process

### PR DESCRIPTION
**Pull request by CRZbulabula**
Reinforce the startup process of Seed-ConfigNode and Non-Seed-ConfigNode, see these docs for implementation details:

1. [Seed-ConfigNode](https://apache-iotdb.feishu.cn/docs/doccniPRgxESvXlekL0PuSiHolf#VmgRgq)
2. [Non-Seed-ConfigNode](https://apache-iotdb.feishu.cn/docs/doccniPRgxESvXlekL0PuSiHolf#9YF2vn)

These cases have been self-tested:

1. Start 3C, success
2. Randomly kill 1C then restart, success
3. Kill 3C then restart, success

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java [hunk 1]
```diff
@@ -83,26 +83,39 @@ public TConfigNodeRegisterResp registerConfigNode(
...
         LOGGER.warn(
             "Add Consensus Group failed because {}, retrying {} ...", e.getMessage(), retry);
         doRetryWait(retry);
       }
     }
-    LOGGER.error("Add ConsensusGroup failed", lastException);
-    return new TSStatus(TSStatusCode.ALL_RETRY_FAILED.getStatusCode())
-        .setMessage("All retry failed due to " + lastException.getMessage());
+
+    throw lastException;
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java:105]: does not throw exception  in last code

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java [hunk 2]
```diff
@@ -93,7 +85,7 @@ public class NodeInfo implements SnapshotProcessor {
 
   // Online DataNodes
   private final ReentrantReadWriteLock dataNodeInfoReadWriteLock;
-  private final AtomicInteger nextNodeId = new AtomicInteger(1);
+  private final AtomicInteger nextNodeId = new AtomicInteger(0);
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:88]: why is 0?   if isseednode , nodeid is 0.

**Review by qiaojialin — APPROVED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:88]: The Seed-ConfigNode will apply itself

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**